src/sites_toolbox.py

```diff
@@ -372,3 +403,41 @@ def sites_toolbox_page():
                     st.json(site_details)
                     if st.button("Close site details"):
                         st.empty()
+
+    # create a new user
+    st.markdown(
+        f'<h1 style="color:#63BCAF;font-size:32px;">{"Create New User"}</h1>',
+        unsafe_allow_html=True,
+    )
+
+    with st.expander("Input new user data"):
+        with connection.get_session() as session:
+            st.markdown(
+                f'<h1 style="color:#FFD053;font-size:22px;">{"User Information"}</h1>',
+                unsafe_allow_html=True,
+            )
+            email = st.text_input("User Email")
+            site_group = st.selectbox("Select a group", site_groups, key="site_group")
+            email_validation = validate_email(email)
+            # check that site group exists
+            if st.button(f"Create new user"):
+                if email_validation is False:
+                    st.markdown(f'<p style="color:#f07167;font-size:16px;">{"Please enter a valid email address."}</p>',
+                                unsafe_allow_html=True,)
+                else:
+                    site_group = get_site_group_by_name(
+                        session=session, site_group_name=site_group
+                    )
+                    user = create_user(
+                        session=session,
+                        email=email,
+                        site_group_name=site_group.site_group_name,
```

Comment:
Seems like you have the `site_group_name`, then you get the site_group, and then get the name from it. 

perhaps further up `site_group` should be renamed `site_group_name` to make it clear

why dont use just pass `site_group_name` into this, seems

Reply:
i think there's something funny with an SQL Alchemy object being returned and throwing an error, but I should be able to clean up the code so it's clearer.

Reply:
ok, let me know if you need any help on that